runtime/lua/vim/_defaults.lua

```diff
@@ -107,22 +107,50 @@ do
           vim.inspect(cmd.cmd)
         )
       end
+      return err
+    end
 
-      if err then
-        vim.notify(err, vim.log.levels.ERROR)
+    local function do_open_shotgun()
```

Comment:
nit: we could model this as a `get_url` function, which may later find its way into a util module or even the stdlib, which may later be used to enhance the behavior of `<cfile>`. 
```suggestion
    local function get_url()
```